src/codeflare_sdk/cluster/config.py

```diff
@@ -69,9 +114,45 @@ def __post_init__(self):
 
         self._memory_to_string()
         self._str_mem_no_unit_add_GB()
+        self._old_gpu_arg_conversion()
         self._memory_to_resource()
-        self._gpu_to_resource()
         self._cpu_to_resource()
+        self._gpu_to_resource()
+        self._combine_custom_resource_mapping()
+
+    def _combine_custom_resource_mapping(self):
+        if overwritten := set(self.custom_resource_mapping.keys()).intersection(
+            DEFAULT_RESOURCE_MAPPING.keys()
+        ):
+            if self.overwrite_default_resource_mapping:
+                warnings.warn(
+                    f"Overwriting default resource mapping for {overwritten}",
+                    UserWarning,
+                )
+            else:
+                raise ValueError(
+                    f"Resource mapping already exists for {overwritten}, set overwrite_default_resource_mapping to True to overwrite"
+                )
+        self.custom_resource_mapping = {
+            **DEFAULT_RESOURCE_MAPPING,
+            **self.custom_resource_mapping,
+        }
+
+    def _gpu_to_resource(self):
+        if self.num_head_gpus:
+            if "nvidia.com/gpu" in self.head_extended_resource_requests:
```

Comment:
This would happen if a user specified:
ClusterConfig(num_gpus=1, worker_extended_resources={"nvidia.com/gpu": 1}).

Just preventing the user from trying to specify the num gpus twice until the num_gpus arg is fully deprecated